### PR TITLE
feat: add webcam selection

### LINKED ISSUE #15: Cannot select webcam source.
I have multiple webcams. Looks like the code shows only the first one (which isn't the right one). Need to be able to pick.

### PR DESCRIPTION
**Pull request by Killusions**
Closes #15

Adds webcam selection. Due to browser limitations this does not work locally.

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -19,6 +19,22 @@
...
+              for (var i = 0; i < vals.length; i++) {
+                selectOption = document.createElement("option");
+                selectOption.innerText = vals[i];
+                selectOption.value = vals[i];
+                cameraSelect.appendChild(selectOption);
+              }
+		}
+		function setCameraSelection(val){
+			document.getElementById('camera').value = val;
+		}
```
Comment (nhtua) [inline index.html:37]: Could you move these 2 functions to /app.js

Comment (nhtua) [inline index.html]: Could you move these function to /app.js ?

File: app.js [hunk 2]
```diff
@@ -78,9 +78,23 @@ function camSizeChange(width, height) {
 var loadParams = getLoadParams();
 var modelParams = getModelParams();
 
+function setCameraDevices(devices) {
```
Comment (nhtua) [inline app.js:81]: I'm very confused the usage of 4 functions:
1. setCameraDevices()
2. setCameraOptions()
3. setCameraSelection()
4. setCamera()

you may merge [1,2] and [3,4]

**Review by nhtua — CHANGES_REQUESTED** [review]
Thank you. This is a great feature which many people are waiting for. Just have a small request to make code clearer/

File: app.js (hunk 2 above)
Comment (Killusions) [reply app.js:81]: Yeah makes sense, done.

**Comment by Killusions** [comment]
@nhtua Thanks for the quick answer, I fixed your issue.

**Comment by Killusions** [comment]
@nhtua I also added an option for an alternative camera, as in a system with a ton of virtual cams (like mine), it's necessairy in case the main camera is for example unplugged.

**Comment by Killusions** [comment]
@nhtua Any issues on this? I cannot view the changes you requested because I pushed the commits again.

**Comment by nhtua** [comment]
Hey @Killusions Thanks for reminding me. Just checked the PR, code looks neat now. But I found a small issue. Drop list only has one `default` value. It only updates the full Camera list after I hit the button `Reload`. 
I have only one Camera in my system, so I can't check what if it works with `default` values or not. Could you help to verify? if it works as expected, please infirm me. I'm happy to merge.

**Comment by nhtua** [comment]
Anyways, thanks for your help on this. You're superb!

**Comment by Killusions** [comment]
> Hey @Killusions Thanks for reminding me. Just checked the PR, code looks neat now. But I found a small issue. Drop list only has one `default` value. It only updates the full Camera list after I hit the button `Reload`.
> I have only one Camera in my system, so I can't check what if it works with `default` values or not. Could you help to verify? if it works as expected, please infirm me. I'm happy to merge.
> 
> 

@nhtua Thank you for your review. I could reproduce this when the camera permission is not yet given. I pushed a simple fix for it.

**Review by nhtua — APPROVED** [review]
Looks good!

**Pull request merged**